src/lm_buddy/jobs/configs/prometheus.py

```diff
@@ -0,0 +1,42 @@
+from typing import Literal
+
+from pydantic import conlist, model_validator
+
+from lm_buddy.types import BaseLMBuddyConfig
+from lm_buddy.jobs.configs import LMBuddyJobConfig
+from lm_buddy.integrations.wandb import WandbRunConfig
+from lm_buddy.integrations.vllm import InferenceServerConfig
+from lm_buddy.integrations.huggingface import TextDatasetConfig, AutoTokenizerConfig
+
+class PrometheusCompletionsConfig(BaseLMBuddyConfig):
+    """Configuration for a "local-completions" prometheus model.
+
+    The prometheus model is powered by a self-hosted inference server, specified
+    as an `InferenceServerConfig`. Additional arguments are also provided
+    to control the tokenizer type and generation parameters.
+    """
+
+    inference: InferenceServerConfig
+
+    # vLLM-served model params
+    best_of: int = 1
+    max_tokens: int = 512 
+    frequency_penalty: float = 1.03
+    temperature: float = 1.0
+    top_p: float = 0.9
+
+    # evaluation script params
```

Comment:
This section could become its own config, while the previous one could be general for all vLLM-served models. WDYT?

Reply:
I think it depends. Which of these parameters are general for _any_ calls to a vLLM inference server, and which are specific to the Prometheus code? 

It seems like some of these are the general HuggingFace inference parameters (top_k, repetition_penalty, etc..). Perhaps those should exist on their own config like `GenerationConfig` or something?

Reply:
Exactly, that was my thought. All of the params in that section are vLLM params (which, despite serving the same HF models, might differ from other inference servers depending on the API they implement).
So my thought was that we could have these (and potentially other) parameters for a vLLMGenerationConfig and import it in the Prometheus-specific config. The only thing we miss is the defaults (I liked the idea of having the values used by kaistai's team as defaults here), but I think it's ok - we can just provide a ready-to-use configuration file with those defaults in.

Reply:
> The only thing we miss is the defaults

I don't follow, why would we be missing the defaults if we created a `vLLMGenerationConfig` class? We could set the defaults there.

Anyhow, I think having those parameters bundled into a re-usable Pydantic model makes sense.

Reply:
What I meant is if we have a `vLLMGenerationConfig` and we assume to use it generally for vLLM models, then it is better not to define it with Prometheus' specific defaults. But that's ok, I'll go with the reusable model and an explicit config